olp-cpp-sdk-dataservice-read/src/VolatileLayerClientImpl.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <memory>
+
+#include <olp/core/client/ApiError.h>
+#include <olp/core/client/ApiResponse.h>
+#include <olp/core/client/CancellationToken.h>
+#include <olp/core/client/HRN.h>
+#include <olp/core/client/OlpClientSettings.h>
+#include <olp/dataservice/read/DataRequest.h>
+#include <olp/dataservice/read/model/Data.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+/**
+ * @brief The VolatileLayerClientImpl aimed to acquire data from volatile layer
+ * of OLP. A volatile layer is a key/value store where values for a given key
+ * can change and only the latest value is retrievable.
+ */
```

Comment:
Do we need to copy the same doxygen comments for `VolatileLayerClientImpl`?

I mean it is private class and not available for the user. So, I suggest to remove doxygen comments for ctor and other public methods of this class. Otherwise, we have to maintain 2 copies of doxygen comments for 2 files. Thus, there is a chance that they will be misaligned, thus will just confuse the reader.

Reply:
Agree